File: src/python/_1921_Eliminate_Maximum_Number_of_Monsters.py
class Solution:
    def eliminateMaximum(self, dist: List[int], speed: List[int]) -> int:
        time = sorted([d / s for d, s in zip(dist, speed)])

        for i, t in enumerate(time):
            if i >= t:
                return i

        return len(dist)

File: src/_1921_Eliminate_Maximum_Number_of_Monsters.java
class Solution {
    public int eliminateMaximum(int[] dist, int[] speed) {
        double[] time = new double[dist.length];

        for (int i = 0; i < dist.length; i++)
            time[i] = (double) dist[i] / speed[i];

        Arrays.sort(time);

        for (int i = 0; i < dist.length; i++)
            if (i >= time[i])
                return i;

        return dist.length;
    }
}
